x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -74,17 +71,26 @@ interface RawModuleConfig {
   };
 }
 
+function isModule(arg: any): arg is Module {
+  return arg.jobs && arg.jobs.length && arg.jobs[0].config !== undefined;
```

Comment:
this type guard seems unsafe... shall we try the following? 
```suggestion
  return typeof arg === 'object' && arg !== null && Array.isArray(arg) && arg.jobs[0]?.config !== undefined;
```

Reply:
https://github.com/elastic/kibana/pull/92855/commits/961396d595104dfe9d633151737a81fde9764366